### ISSUE TITLE
Add ability to scope annotations

### ISSUE THREAD

**Issue by russorat** [influxdata/chronograf#3143]
use case: i have a bunch of hosts being monitored, and i would like to annotate things that happen on this host, but those events are not relevant to other hosts.

In this case, i need to be able to define annotations in such a way that they only show up on graphs with this host.

**Comment by alexpaxton** [comment]
### Design

Instead of entering "Annotate" mode via a cell's edit menu, the user will click the `Annotate` button in the dashboard header. The feature behaves differently depending on whether the Template Variables Control Bar is visible or not.

If the Template Variables Control Bar is hidden the new annotation will be scoped globally.
This is reflected in the new annotation tooltip:
<img width="382" alt="screen shot 2018-04-24 at 5 36 35 pm" src="https://user-images.githubusercontent.com/2433762/39221706-79de759a-47ed-11e8-8591-bae98303dd85.png">

If the Template Variables Control Bar is visible then the new annotation assumes the scope of the currently set template variable values. In this mode the new annotation crosshair and tooltip are purple to indicate the difference. There is also another line in the tooltip which lists the template variable values:
<img width="762" alt="screen shot 2018-04-24 at 5 55 56 pm" src="https://user-images.githubusercontent.com/2433762/39221754-bbd78720-47ed-11e8-828b-9b73b250b9da.png">

There will no longer be cell menu items relating to annotations:
![screen shot 2018-04-24 at 5 56 53 pm](https://user-images.githubusercontent.com/2433762/39221773-d5268b5e-47ed-11e8-9b49-b543610aa915.png)

There will no longer be a "Edit Annotations" mode which means no more `Done` buttons appearing on multiple cells. To edit an annotation the user will click an `Edit` button from the tooltip directly:
![screen shot 2018-04-24 at 6 02 39 pm](https://user-images.githubusercontent.com/2433762/39221815-0a719a42-47ee-11e8-9c5c-20923afb0ec7.png)

This will open an overlay specifically for the annotation:
![image](https://user-images.githubusercontent.com/2433762/39221889-5ff6f6ce-47ee-11e8-92ce-295a299c7aa3.png)

The user can edit the Annotation's name and timestamp here. Manually typing a timestamp is more precise and powerful than manually repositioning the annotation.

If the annotation has a template variable scope it appears like so:
![image](https://user-images.githubusercontent.com/2433762/39221893-641e7376-47ee-11e8-9d90-520bed90afb0.png)

The user can remove individual key/value pairs from the scope and cannot add them. It will be faster to delete the annotation and re-draw it. Deleting all the key/value pairs sets the annotation scope to `global`

If the timestamp contains both start and end points the `Timestamp` field is split into two fields:
![image](https://user-images.githubusercontent.com/2433762/39221971-d156f8be-47ee-11e8-8842-fdffe475fcf0.png)

**Comment by russorat** [comment]
I really like this! 😀

**Comment by russorat** [comment]
- [ ] break out the api work into a separate story

**Comment by nhaugo** [comment]
@alexpaxton if there is a popup that opens for editing the annotation then why not just add an option to add to the current cell (default), a multi select for a list of cells, all cells on the dashboard or global.

I can not think of a single use case for global annotations, so this should never be a UX default.

**Comment by russorat** [comment]
@nhaugo @chnn the idea here is that the annotation should be connected to the data somehow, not just the cell. If it's only applied to a cell, you miss the opportunity for correlating different metrics across dashboards.

For a concrete example, if i add an annotation to a particular cluster that it was restarted. I have multiple dashboards for monitoring stats on my cluster. I want the annotation to show up in all the dashboards when I filter for that specific cluster.

Also, if I delete and recreate the cell, i still want the annotations to be displayed for that cluster.

**Comment by chnn** [comment]
connect #3905

**Comment by russorat** [comment]
@alexpaxton see #3905 for how we want to do this. let's work on updated ux to support this.

**Comment by alexpaxton** [comment]
![screen shot 2018-07-17 at 11 22 44 am](https://user-images.githubusercontent.com/2433762/42837748-c4d6fb3c-89b3-11e8-80b7-a56640f424bb.png)

Control bar placement in Data Explorer

- Same `Tools` toggle as in dashboard header
- Use same key in local storage for determining whether to show or hide

**Comment by alexpaxton** [comment]
[scoped-annotations--v1.pdf](https://github.com/influxdata/chronograf/files/2224740/scoped-annotations--v1.pdf)

Updated designs that center around creating filters (similar to log viewer UX) and assigning key/value pairs to annotations

**Comment by russorat** [comment]
@chnn what is this blocked on?

**Comment by chnn** [comment]
@russorat just the API now, working on it in #4072
